Assets/MRTK/SDK/Experimental/StepSlider/StepSlider.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.UI;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.UI
+{
+    [AddComponentMenu("MRTK/SDK/Experimental/StepSlider")]
+    public class StepSlider : PinchSlider, IMixedRealityPointerHandler
+    {
+
+        #region Serialized Fields and Properties
+        [Experimental]
+        [SerializeField]
+        [Tooltip("Number of subdivisions of slider values.")]
+        
+        private int sliderStepDivisions = 0;
+        /// <summary>
+        /// Property accessor of sliderStepDivisions, it holds the number of subdvisions the slider is split into.
+        /// </summary>
+        public int SliderStepDivisions
+        {
+            get 
+            { 
+                return sliderStepDivisions; 
+            }
+            set
+            {
+                sliderStepDivisions = value;
+            }
+        }
+        #endregion
+
+        #region Protected Members
+        /// <summary>
+        /// Private member used to adjust slider values
+        /// </summary>
+        protected float sliderStepVal = 0.01f;
+        #endregion
+
+        #region Unity methods
+        new void Start()
+        {
+            updateStepVal();
+            base.Start();
+        }
+        #endregion
+
+        #region Private Methods
+        /// <summary>
+        /// Private method used to calculate and update the slider step divisions based on the number provided 
+        /// </summary>
+        private void updateStepVal()
+        {
+            if (SliderStepDivisions >= 1)
+            {
+                var startVal = 0.0f;
+                var endVal = 1.0f;
+                sliderStepVal = (endVal - startVal) / sliderStepDivisions;
+                SliderValue = (startVal + (sliderStepDivisions / 2) * sliderStepVal);
```

Comment:
This seems to be called on start only, meaning it will adjust the slider value to the middle of the slider track on start. What if the user wants the slider value to start at say, 0.1 on a slider ranging from 0 to 1, with 10 steps? Is there a way to do this? I would expect if I set slidervalue in editor to 0.1, that when I hit play, it would stay at 0.1.